src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,62 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  padding: 10px;
+  width: $calendar-width;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid #000;
+    background-color: $day-color;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $day-color-hover;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+@for $day-number from 1 through 31 {
+  .calendar__day--#{$day-number} {
+    &::before {
+      content: '#{$day-number}';
+    }
+  }
+}
+
+$week-days: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+
+@each $week-day, $index in $week-days {
+  .calendar--start-day-#{$week-day} :first-child {
+    margin-left: $index * ($day-size + $day-gap);
```

Comment:
The variable `$day-size` is used here in a calculation but is not defined within this file. Ensure that it is defined or imported from another file to avoid compilation errors.